### DISCUSSION TITLE
Question about WEBSITE_RUN_FROM_PACKAGE and bandwidth use

### DISCUSSION THREAD

**Discussion by sg3-141-592** [polymind-inc/acmebot#632] [Q&A]
Hey there Shibayan, quick question on your project.

I've got a new piece of self-deployed open source and I've tried to use a similar model to keyvault-acmebot of using WEBSITE_RUN_FROM_PACKAGE to control the deployment of the app https://github.com/sg3-141-592/AzStartStop . Python bundles are a bit bigger than C# (26MB bundle) and the bandwidth costs are killing me. I'm at 20GB a day for a fairly modest user base, and every time the function is called it's another 26MB of bandwidth used.

Do you know of any detail I might be missing to get my bandwidth under control. Or do you just have to provide a very large amount of storage account bandwidth to support the keyvault-acmebot users?

**Comment by shibayan** [comment]
Since I deliver packages from Azure Storage, I incur transfer costs if I am not in the same region, but fortunately I receive Azure sponsorship for open source so I am able to operate.

If you use GitHub Release assets, you will not incur any transfer costs, so you may want to consider using those first.

**Comment by sg3-141-592** [comment]
Thanks for the quick response and suggestion. Let me have a look at the rate limits on GitHub Releases, see if it's practical. It would be a quick way to fix.

Otherwise if that doesn't work I'm testing including a deployment script in the ARM template to copy the latest release into blob storage. Not ideal but might be workable.

**Comment by sg3-141-592** [comment]
Hey @shibayan , just for interest. I've developed an ARM template that copies a deployment off a specified url into blob storage and returns a Sas token to it for use in WEBSITE_RUN_FROM_PACKAGE https://gist.github.com/sg3-141-592/15e2cbb67694d9b3c1fda9615db5f199 .

Not very elegant but best workaround I've found so far.
